projects/plugins/jetpack/extensions/plugins/payments/payments.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * Payments plugin.
+ *
+ * @since $$next-version$$
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\Payments;
+
+use Automattic\Jetpack\Blocks;
+use Jetpack;
+
+const FEATURE_NAME = 'payments';
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+
+/**
+ * Registers the plugin for use in the block editor's plugin sidebar.
+ * This is done via an action so that we can disable registration if we need to.
+ */
+function register_plugin() {
+	if (
+		( defined( 'IS_WPCOM' ) && IS_WPCOM )
+		|| Jetpack::is_connection_ready()
+	) {
+		Blocks::jetpack_register_block(
+			BLOCK_NAME,
+			array(
+				'plan_check' => true,
+			)
+		);
+	}
+}
+add_action( 'init', __NAMESPACE__ . '\register_plugin' );
```

Comment:
This looks a bit weird to me: we are registering an editor plugin as a block, and I'm not sure it's the right way to go — unless I'm overlooking something.

For example, the Publicize plugin too extends the publish panel (the pre-publish one, but anyway), and doesn't register as a block:

https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/extensions/plugins/publicize/publicize.php

Reply:
That's a good point. I'm happy to change that. Done in 9b50bb0a5c99a7ec784913481e81b844e35cb50e